src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -423,40 +429,67 @@ namespace AppInstaller::Settings
         return UserSettings::Instance();
     }
 
-    UserSettings::UserSettings() : m_type(UserSettingsType::Default)
+    bool TryInitializeCustomUserSettings(std::string content)
+    {
+        if (s_userSettingsInitialized || s_userSettingsInInitialization)

```

Comment:
This is probably the right approach for now, but we should have a test to make sure that our suggested method for initialization is working so that we don't load the settings earlier than that.

Reply:
yes, I verified manually it's working as expected, and it should and will be part of Com e2e test.